flocker/node/functional/test_deploy.py

```diff
@@ -405,11 +407,13 @@ def test_cpu_shares(self):
         def inspect_application(_):
             deferred_list = docker_client.list()
 
-            def app_memory(units):
-                unit = units.pop()
-                self.assertEqual(unit.cpu_shares, EXPECTED_CPU_SHARES)
-                return deferred_list
-
-            deferred_list.addCallback(app_memory)
+            def app_cpu_shares(units):
+                checked = [
+                    self.assertEqual(unit.cpu_shares, EXPECTED_CPU_SHARES)
+                    for unit in units if unit.name == application_name
+                ]
+                self.assertNotEqual(checked, [])
```

Comment:
How about a shared callback for filtering `units` down to one matching a particular name?  `deferred_list.addCallback(pick_unit_named, application_name); deferred_list.addCallback(app_cpu_shares)`

Avoids some duplication between these two tests.